### PR TITLE
feat: Data types

### PR DESCRIPTION
**Pull request by AnnDv**

### CODE REVIEW

File: dataTypes/arrayMethods.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//sorts by age
+function sortByAge(arr) {
+    arr.sort((a, b) => a.age - b.age);
+  }
+  
+  let john = { name: "John", age: 25 };
+  let pete = { name: "Pete", age: 30 };
+  let mary = { name: "Mary", age: 28 };
+  
+  let arrayNames = [ pete, john, mary ];
```
Comment (chitanumihai) [inline dataTypes/arrayMethods.js:26]: [ pete, john, mary ];

spaces in array?

File: dataTypes/string.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    let toUpp = str[0].toUpperCase() + str.slice(1)
+    return toUpp
+}
+
+console.log(ucFirst("hellen"))
+
+// checks for spam
+function checkSpam(str) {
+    let strToLower = str.toLowerCase()
+    if (strToLower.includes("viagra") || strToLower.includes("xxx")) {
```
Comment (chitanumihai) [inline dataTypes/string.js:12]: lets stick to mock data like: "John Doe", "Alissa bounty"

File: classes/classes.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+//rewrites clock functional style in the "class" syntax
+class Clock {
+    constructor({ template }) {
+      this.template = template;
+    }
+  
+    render() {
+      let date = new Date();
```
Comment (chitanumihai) [inline classes/classes.js:8]: why let and not const?

File: dataTypes/arrayMethods.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// sorts in decreasing order
+function sortDeacreased(arr) {
+    return arr.sort((a, b) => b - a);
+}
+
+let array = [3, 5, 1 , 8, 9, 4];
```
Comment (chitanumihai) [inline dataTypes/arrayMethods.js:6]: let let let

File: dataTypes/string.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// returns the string with uppercased first character
+function ucFirst(str){
+    let toUpp = str[0].toUpperCase() + str.slice(1)
+    return toUpp
```
Comment (chitanumihai) [inline dataTypes/string.js:4]: dont forget about semicolons ;

File: dataTypes/arrays.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+let styles = ["Jazz", "Blues"];
+styles.push("Rock-Roll");
+styles[Math.floor((styles.length - 1) / 2)] = "Classics";
+alert( styles.shift() );
+styles.unshift("Rap", "Reggae");
+
+//sum input numbers
+function sumInput() {
+
+    let numbers = [];
```
Comment (vadimb) [inline dataTypes/arrays.js:11]: Try to switch to const by default

**Review by vadimb — APPROVED** [review]

**Pull request merged**